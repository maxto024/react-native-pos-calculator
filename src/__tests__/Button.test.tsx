import { shallow } from 'enzyme'
import * as React from 'react'
import { Button } from '../Button'

it('renders button correctly', () => {
  const spy = jest.fn()
  const wrapper = shallow(<Button text="foo" onPress={spy} />)
  expect(wrapper).toMatchSnapshot()

  // set props tests
  wrapper.setProps({
    text: 'bar',
    style: {
      borderStyle: 'solid',
      borderRightWidth: 1,
      borderBottomWidth: 1
    },
    textStyle: { color: '#fff', fontSize: 1 }
  })
  expect(wrapper).toMatchSnapshot()

  // press button test
  wrapper
    .first()
    .props()
    .onPressIn()
  expect(spy).toBeCalled()
})
