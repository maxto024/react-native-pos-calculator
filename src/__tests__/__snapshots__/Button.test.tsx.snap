// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPressIn={[MockFunction]}
  style={
    Array [
      undefined,
    ]
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      foo
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`renders button correctly 2`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPressIn={[MockFunction]}
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderRightWidth": 1,
        "borderStyle": "solid",
      },
    ]
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 1,
        }
      }
    >
      bar
    </Text>
  </Component>
</TouchableOpacity>
`;
